mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusTest.java

```diff
@@ -481,10 +515,7 @@ void dispatchShouldWorkAfterNetworkIssuesForNewKeyRegistration() {
                 when(listener.getExecutionMode()).thenReturn(MailboxListener.ExecutionMode.ASYNCHRONOUS);
 
                 rabbitMQExtension.getRabbitMQ().pause();
-
-                assertThatThrownBy(() -> eventBus.dispatch(EVENT, NO_KEYS).block())
-                    .isInstanceOf(RabbitFluxException.class);
-
+                eventBus.dispatch(EVENT, NO_KEYS).block();
```

Comment:
If dispatching is unsuccessful, then as a caller I need to know it, right?

Here we are loosing events, and data-loss is not OK...

Returning a failure would allow the caller to also report this failure and ultimately report/log the error. Here I fear stuff go un-noticed.

There is not event a strategy in place to mitigate this behaviour - storing such an event in deadLetter ?-.

Reply:
I'm agree, what about you @mbaechler ?

Reply:
The only drawback of the new code is that it creates a window where the push of the event in rabbit fails and the code trying to push is not aware of it.
The push is blocking the client until the "SynchronousQueue to Flux" code is calling remove on the queue. So the only possible failure is the "SynchronousQueue to Flux" code failing to call sender.send on rabbit.
```java
sender.send(Mono.fromCallable(eventSender::take).repeat().subscribeOn(Schedulers.elastic()))
            .doOnError(throwable -> LOGGER.error("Cannot send event to rabbitMQ", throwable))
            .subscribe()
```
The right way to solve that would be to block the calling code until `send` actually completes and return an error if it doesn't.
I didn't managed to do that.
I propose to try to solve that tiny issue into another PR and have this one merge (because it solves big resource consumption issue). WDYT?